paasta_tools/mesos_tools.py

```diff
@@ -570,6 +570,7 @@ def get_mesos_slaves_grouped_by_attribute(slaves, attribute):
     }
 
 
+@time_cache(ttl=10)
```

Comment:
im not so keen on this, since it isn't really 'opt in'. can you change the caller instead?

Reply:
To change the caller, I'll need to rewrite [mesos_tools.get_all_slaves_for_blacklist_whitelist](https://github.com/Yelp/paasta/blob/master/paasta_tools/mesos_tools.py#L538), and then [smartstack_tools.get_smartstack_replication_for_attribute](https://github.com/Yelp/paasta/blob/master/paasta_tools/smartstack_tools.py#L140) and [marathon_serviceinit.status_smartstack_backends](https://github.com/Yelp/paasta/blob/master/paasta_tools/marathon_serviceinit.py#L284) because they use get_all_slaves_for_blacklist_whitelist and so on.

I'm happy to do this, because I don't feel we should have a function that silently query mesos master every time we call it, but could you confirm please that we are on the same page regarding changing the caller?

Reply:
This PR is not complete yet, but I've already changed what you asked.